lib/http.js

```diff
@@ -0,0 +1,33 @@
+const http = require('http')
+const https = require('https')
+
+const getRequest = (url, timeout = 30 * 1000) =>
+  new Promise((resolve, reject) => {
+    const urlObject = new URL(url)
+    const httpRequestLib = urlObject.protocol === 'https:' ? https : http
+    const httpRequest = httpRequestLib.get(urlObject, { timeout }, res => {
+      const rawData = []
+
+      res.setEncoding('utf8')
+      res.on('data', chunk => rawData.push(chunk))
+      res.on('end', () => {
+        if (res.statusCode < 200 || res.statusCode >= 300) {
+          const errorMsg = res.body || res.statusMessage || `Http Error ${res.statusCode}`
+          reject(errorMsg)
+        } else {
+          try {
+            const data = rawData.join('')
+            resolve(data && JSON.parse(data))
```

Comment:
Thanks for updating this.  I think it's fine to assume the content-type for the get requests is JSON here since we're just using it for the call to the `keys` endpoint, which we know is JSON per the spec.  In the off-chance that `data` is an empty string, do we want to be `resolving` the promise with that empty string or rejecting it with an error that makes it clear what happened.

Reply:
Of course, that makes sense to me 👍🏽 . Updated it here: https://github.com/articulate/authentic/pull/30/commits/3688b4ae8e7b942386d244b8e3182b934eaf022e.